### ISSUE TITLE
Subscription.Update() method does not allow to update subscription ID

### ISSUE THREAD

**Issue by vania-pooh** [lionelbarrow/braintree-go#242]
Just compare. In Java client it works like the following:
```
SubscriptionRequest request = new SubscriptionRequest()
  .id("new_id")
  .paymentMethodToken("new_payment_method_token")
  .price(new BigDecimal("14.00"))
  .planId("new_plan")
  .merchantAccountId("new_merchant_account");

Result<Subscription> result = gateway.subscription().update(
  "a_subscription_id",
  request
);
```
In this Go client the value from `SubscriptionRequest` is always used. This prevents user from updating the ID. So I think a correct signature should be `Update(ctx context.Context, id string, sub *SubscriptionRequest)`.

**Comment by vania-pooh** [comment]
As usually can send a fix - but let's first of all discuss how to do this better. Should I break current API by changing method signature or just add one more method?

**Comment by vania-pooh** [comment]
@leighmcculloch @lionelbarrow any comments on this?

**Comment by leighmcculloch** [comment]
Hi @vania-pooh! I've been on vacation and not at a computer, so I'm just seeing this now. I agree, looks like the API is missing the ability to update the ID. The Braintree docs even provide an example of being able to update an ID so we should try and support it too.

Ref: https://developers.braintreepayments.com/reference/request/subscription/update

👍 if you're up for opening a PR.

**Comment by leighmcculloch** [comment]
I'd prefer the change just break the API.

**Comment by vania-pooh** [comment]
@leighmcculloch ok, will send respective PR soon.

**Comment by vania-pooh** [comment]
@leighmcculloch take a look at my changes in https://github.com/lionelbarrow/braintree-go/pull/243.

**Comment by leighmcculloch** [comment]
Fixed by #243.
